api/cluster/resource/templater.go

```diff
@@ -550,6 +596,120 @@ func createLoggerSpec(loggerURL string, loggerConfig models.LoggerConfig) *kserv
 	}
 }
 
+// createNewInferenceServiceTopologySpreadConstraints creates topology spread constrains for a component of a new
+// inference service
+func createNewInferenceServiceTopologySpreadConstraints(
+	modelService *models.Service,
+	config *config.DeploymentConfig,
+	component kservev1beta1.ComponentType,
+) ([]corev1.TopologySpreadConstraint, error) {
+	if len(config.TopologySpreadConstraints) == 0 {
+		var topologySpreadConstraints []corev1.TopologySpreadConstraint
+		return topologySpreadConstraints, nil
+	}
+	var newRevisionName string
+	if modelService.DeploymentMode == deployment.RawDeploymentMode {
+		newRevisionName = fmt.Sprintf("isvc.%s-%s-default", modelService.Name, component)
+	} else if modelService.DeploymentMode == deployment.ServerlessDeploymentMode ||
+		modelService.DeploymentMode == deployment.EmptyDeploymentMode {
```

Comment:
I've specified the `EmptyDeploymentMode` here too since the existing unit tests seem to suggest that we're expecting the behaviour of a `ServerlessDeploymentMode` if nothing (empty string `""`) is specified as the `DeploymentMode`.